src/xma/src/xmaapi/xmaxclbin.cpp

```diff
@@ -88,192 +84,140 @@ static int32_t kernel_max_channel_id(const ip_data& ip, std::string kernel_chann
   return ctxid;
 }
 
-static int get_xclbin_iplayout(char *buffer, XmaXclbinInfo *xclbin_info)
+static int get_xclbin_iplayout(const std::vector<char> &buffer, XmaXclbinInfo *xclbin_info)
```

Comment:
It would be better if the caller would simply calls 
get_xclbin_iplayout(vec.data(), ...)
so as not to change the signature of this function.  The fact the the buffer is wrapped in a std::vector is of no interest to clients of the buffer.

Reply:
done